plugins/org.wso2.developerstudio.visualdatamapper.diagram/src/org/wso2/developerstudio/datamapper/diagram/custom/configuration/operator/transformers/DirectOperatorTransformer.java

```diff
@@ -61,4 +66,35 @@ public String generateScriptForOperation(Class<?> generatorClass, List<DMVariabl
 		}
 		return operationBuilder.toString();
 	}
+	
+	private String appendTypeCorrectedInputVariable(StringBuilder operationBuilder, List<DMVariable> inputVariables, Map<String, List<SchemaDataType>> variableTypeMap,
+			Stack<ForLoopBean> parentForLoopBeanStack, List<ForLoopBean> forLoopBeanList, Map<String, Integer> outputArrayVariableForLoop,
+			Map<String, Integer> outputArrayRootVariableForLoop, SchemaDataType outputDataType) {
+		
+		try {
+			String prettyVariable = ScriptGenerationUtil.getPrettyVariableNameInForOperation(inputVariables.get(0), variableTypeMap,
+				parentForLoopBeanStack, true, forLoopBeanList, outputArrayVariableForLoop, outputArrayRootVariableForLoop);
+			SchemaDataType inputDataType = inputVariables.get(0).getSchemaVariableType();
+			String typeConvertedPrettyVariable = "";
+			if(!outputDataType.equals(inputDataType)) {
```

Comment:
Formatting issue shall we fix this and other places as well ?

Reply:
Corrected